www/base/src/app/builders/builders.controller.coffee

```diff
@@ -31,16 +31,21 @@ class Builders extends Controller
             all_tags.sort()
             return all_tags
 
-        $scope.tags_filter = $location.search()["tags"]
-        $scope.tags_filter ?= []
-        if not angular.isArray($scope.tags_filter)
-            $scope.tags_filter = [$scope.tags_filter]
+        updateTagsFilterFromLocation = () ->
+            $scope.tags_filter = $location.search()["tags"]
+            $scope.tags_filter ?= []
+            if not angular.isArray($scope.tags_filter)
+                $scope.tags_filter = [$scope.tags_filter]
+
+        updateTagsFilterFromLocation()
 
         $scope.$watch  "tags_filter", (tags, old) ->
             if old?
                 $location.search("tags", tags)
         , true
 
+        $rootScope.$on '$locationChangeSuccess', updateTagsFilterFromLocation
```

Comment:
Do you mind if I report a bug to keep track of it? I don't know whether I have time to properly factorize this in the near future.